### ISSUE TITLE
Activate and deactivate buttons are in wrong state 

### ISSUE THREAD

**Issue by anthonykim1** [microsoft/vscode-python-environments#49]
Testing #24503

First case, the deactivate button persist and does not switch to activate button even after successfully deactivating: 
![Image](https://github.com/user-attachments/assets/50687d54-3889-4b15-a30d-651240076539)

Second case, 
Say I have two separate zsh instances in VS Code terminal:
1. The very first terminal, I click deactivate and leads to successful deactivation, the UI shows activate button as expected.
2. The second zsh instance where environment is activate, it does not show deactivate button but rather activate button. I believe this is wrongly cached? See the video.

https://github.com/user-attachments/assets/a1e8dcd7-7b90-4854-872b-bdd5f975571e

**Comment by karthiknadig** [comment]
@anthonykim1 How was the second terminal activated? If it was not through the extension then it does not know.

**Comment by anthonykim1** [comment]
I see, when you say through extension, you mean environment manager extension, correct?

**Comment by karthiknadig** [comment]
Yes, the env manager UI has inline context item to create terminal.

**Comment by karthiknadig** [comment]
This might be fixed now. Please try again.
